docs/specs/snp-genotyping-vector.md

```diff
@@ -0,0 +1,185 @@
+    --alleles {alleles VCF} \
+    -R {reference sequence} \
+    --out {output VCF} \
+    --genotype_likelihoods_model BOTH \
+    --genotyping_mode GENOTYPE_GIVEN_ALLELES \ 
+    --heterozygosity 0.015 \
+    --heterozygosity_stdev 0.05 \
+    --indel_heterozygosity 0.001 \
+    --downsampling_type BY_SAMPLE \
+    -dcov 250 \
+    --output_mode EMIT_ALL_SITES \
+    --min_base_quality_score 17 \
+    -stand_call_conf 0.0 \
+    -contamination 0.0 \
+    -A DepthPerAlleleBySample \
+    -XA RMSMappingQuality \
+    -XA Coverage \
+    -XA ExcessHet \
+    -XA InbreedingCoeff \
+    -XA MappingQualityZero \
+    -XA HaplotypeScore \
+    -XA SpanningDeletions \
+    -XA FisherStrand \
+    -XA StrandOddsRatio \
+    -XA ChromosomeCounts \
+    -XA BaseQualityRankSumTest \
+    -XA MappingQualityRankSumTest \
+    -XA QualByDepth \
+    -XA ReadPosRankSumTest
+```
+
+Notes: 
+ 
+1. Because we are genotyping a single sample against given alleles, we
+   don't need many of the variant annotations that we would usually
+   ask for when doing discovery. Reducing the number of annotations
+   should reduce file size somewhat.
+
+2. I've set the '-contamination' option to zero (0.0) here rather than
+   0.05 we've used previously because I didn't realise that this
+   option down-samples only alternate alleles (I thought it
+   downsampled both alt and ref alleles), and so setting contamination
+   > 0 will create bias towards reference allele. We will deal with
+   contamination separately.
+
+3. When genotyping given alleles, GATK ignores any variant in the
+   known variants VCF that is not marked as PASS in the FILTER
+   field. However, we want to genotype at all variants in the known
+   variants VCF, regardless of what is in the FILTER column. Therefore
+   the known variants VCF must be created setting FILTER to PASS for
+   all variants to get this to work as desired.
+
+
+### Step: VCF to Zarr conversion
+
+Convert the VCF to Zarr format via the scikit-allel
+[vcf_to_zarr()](https://scikit-allel.readthedocs.io/en/stable/io.html#allel.vcf_to_zarr)
+function. Note the following implementation details:
+
+* Only the ``variants/MQ``, ``calldata/GT``, ``calldata/GQ`` and
+  ``calldata/AD`` fields are required in the output.
+
+* Data types should be as follows: ``variants/MQ``: i1 (single byte
+  signed integer), ``calldata/GT``: i1; ``calldata/GQ``: i1;
+  ``calldata/AD``: i2
+
+* Compressor should be ``zarr.Blosc(cname='zstd', clevel=5,
+  shuffle=-1)``
+
+* Chunk length should be ``2**18``
+
+* ``alt_number`` parameter should be 3 (fix number of alt alleles)
+
+* Data should be grouped by contig (chromosome). This means calling
+  ``vcf_to_zarr()`` once for each contig in the reference genome,
+  specifying the name of the contig as the value of the ``region`` and
+  ``group`` parameters.
+
+Here is an example of the appropriate function call in Python for a
+single contig (this should be called for all contigs in the genome):
+
```

Comment:
For reference, past pipelines used this script for converting from vcf to zarr:  https://github.com/malariagen/legacy_pipelines/blob/master/prod-tools/scripts/vcf_to_zarr.py

There is additional code for ensuring that the files within the zarr have group read/write permissions.  It's handy on the Sanger farm systems, but not necessary as long as the files are created by a production user.

Reply:
Thanks Thuy. Just to note that as of zarr v2.4 it should no longer be necessary to change permissions, these will be set according to users umask.